components/post-context/readme.md

```diff
@@ -0,0 +1,51 @@
+# `PostContext`
+
+The `PostContext` component allows you to customize the post object referenced by and of the components referencing the current post object. They are all prefixed with `Post`.
+
+For example this can be used to build a custom block that gets used inside the core query loop and accesses the passed in post id / post type of that to power the functionality of all the `Post` child components.
+
+## Usage
+
+```js
+import { PostContext, PostTitle } from '@10up/block-components';
+
+function BlockEdit() {
+
+    return (
+        <PostContext postId={42} postType={'page'} isEditable={true} >
+            <PostTitle tagName="h2" className="wp-block-example-hero__title" />
+        </PostContext>
+    )
+}
+```
+
+The `PostContext` component works great with the Core Query Loop block if you want / need to create a custom block to be used within the post template. Any block that gets used inside the query loop can access the current post id, post type, and query id via the block context. These values can then be used within the `<PostContext>` to make them available for any of the post level components nested within.
+
+```js
+import { PostContext, PostTitle } from '@10up/block-components';
+
+function BlockEdit(props) {
+    const { context } = props;
+    const { postId, postType, queryId } = context;
+    const isDescendantOfQueryLoop = Number.isFinite(queryId);
+
+    return (
+        <PostContext postId={postId} postType={postType} isEditable={!isDescendantOfQueryLoop}>
+            <PostTitle tagName="h2" className="wp-block-example-hero__title" />
+        </PostContext>
+    )
+}
+```
+
+*Note: If you enable `isEditable` prop in the `<PostContext>` component for posts that aren't the current post this will actually update the external post. This should be used with caution since it may not be immediately obvious for users.*
```

Comment:
This is great and I love the warning, I wonder if it makes sense to you to have an `onChange` handler for these. I remember I did something similar to this effort (but less organised and only title and description) where the intention was to let people override the used title/description for editorial purposes but not actually the real title one. `onChange` would offer an escape hatch so you can do as you wish.

Feel free to discard this comment if you don't think it makes sense

Reply:
I updated the readme description here a bit. There has been quite some improvement in core for how this gets handled. With this new handling in place I don't think this is as much of a problem anymore. 

However I do think the addition of a way to override the change handler to be able to have block locallized overrides is very cool! I will do that in a follow up PR so we can wrap this initial work up  👍 Great idea 🚀